Sources/UAPusher/ConnectionManager.swift

```diff
@@ -1,16 +1,20 @@
 import Vapor
 import HTTP
 import Foundation
+import TLS
 
-public final class ConnectionMananger {
+public final class ConnectionManager {
+    static let baseUrl = "https://go.urbanairship.com"
+    private let client: ClientProtocol
+    private let config: UAPusherConfig
     
-    let drop: Droplet
-    let baseUrl = "https://go.urbanairship.com"
-    let config: UAPusherConfig
-    
-    public init(drop: Droplet, config: UAPusherConfig) {
-        self.drop = drop
+    public init(clientFactory: ClientFactoryProtocol, config: UAPusherConfig) throws {
         self.config = config
+        self.client = try clientFactory.makeClient(
```

Comment:
Same comment here as in NStack - if we should just save the `ClientFactoryProtocol` reference instead.